Sources/NIOCore/EventLoopFuture+AssumeIsolated.swift

```diff
@@ -119,26 +130,54 @@ struct IsolatedEventLoop {
 
     /// Returns the wrapped event loop.
     @inlinable
-    func nonisolated() -> any EventLoop {
+    public func nonisolated() -> any EventLoop {
         self._wrapped
     }
 }
+
 extension EventLoop {
     /// Assumes the calling context is isolated to the event loop.
-    @usableFromInline
-    func assumeIsolated() -> IsolatedEventLoop {
-        IsolatedEventLoop(self)
+    @inlinable
+    @available(*, noasync)
+    public func assumeIsolated() -> NIOIsolatedEventLoop {
+        self.preconditionInEventLoop()
+        return NIOIsolatedEventLoop(self)
+    }
+
+    /// Assumes the calling context is isolated to the event loop.
+    ///
+    /// This version of ``EventLoop/assumeIsolated()`` omits the runtime
+    /// isolation check in release builds. It retains it in debug mode to
+    /// ensure correctness.
+    @inlinable
+    public func assumeIsolatedUnsafeUnchecked() -> NIOIsolatedEventLoop {
```

Comment:
Same here. Can we make this `noasync`

Reply:
I deliberately didn't do that. This is already marked unsafe-unchecked. I have therefore left it for the possible use-case that someone is actually trying to do the thing we're trying to prevent them doing, and is willing to take on the risk of doing it correctly themselves.

Reply:
Okay fair enough. Can we add that to the documentation then.